src/app/payment/paymentMethod/DigitalRiverPaymentMethod.tsx

```diff
@@ -21,7 +21,17 @@ const DigitalRiverPaymentMethod: FunctionComponent<DigitalRiverPaymentMethodProp
     ...rest
 }) => {
     const { setSubmitted } = useContext(FormContext);
-    const containerId = `${rest.method}-component-field`;
+    const containerId = `${rest.method.id}-component-field`;
+    let disabledPaymentMethodsFromSettings: string[] = [];
```

Comment:
Could this now be something like:

`const disabledPaymentMethods = rest.method.initializationData?.disabledPaymentMethods;` ?

The value should end up as your array of strings from the API or undefined, a simpler way of writing lines 26 - 33.

And then you should be able to just pass `disabledPaymentMethods` directly as the useCallback requirement.